### DISCUSSION TITLE
playing a PlayList

### DISCUSSION THREAD

**Discussion by t-m-z** [bluefireteam/audioplayers#1360] [Q&A]
I would like to use the Audioplayer to play a list of MP3-Files.
Hence I have to iterate over the list.
My idea is to check the PlayerState and if the player is not playing I increase the index and play the next song.
However this does not work.

I would appreciate a hint.

Kind regards

The code looks like
...
  Future<void> getPlayerState() async {
    setState(() => state = player.state);
  }
...
  PlayerState state;
  PlayerState PLAYING = PlayerState.playing;
...
IconButton(
          icon: Icons.play_arrow,
          onPressed: () {
            index = 0;
              while (index < _playlist.length) {
                if (getPlayerState() != PLAYING ) {
                  player.play(_playlist[index]);
                  index++;
                }
              }
);
...

**Comment by Gustl22** [comment]
You only play the first song, as the `play` statement does not wait until the song is finished. I would recommend to use the `player.onComplete` callback to start a new track, when the current one has finished, without a while loop.

**Reply by t-m-z** [reply]
Thanks for this hint. I tried to understand the doc but failed. However, I found a code snippet from Walter — 27.06.2022 17:28 on your discord server which helped me:
  https://discord.com/channels/509714518008528896/533299043686940692/threads/990854105394216970
